### DISCUSSION TITLE
Pretty Print only works on "httpyac.responseViewContent": "body"

### DISCUSSION THREAD

**Discussion by chrismalek** [AnWeber/vscode-httpyac#55] [Q&A] [answered]
Hello,

First thanks for all the work on this. This is really a stellar plugin. 

I have been watching httpyac closely. I used rest client all day everyday. However, I like some of the subtle features of httpyac and have been wanting to switch over..

I tried to make a switch today to httpyac but the pretty print only seems to work with this setting.

```json
{
"httpyac.responseViewContent": "body"
}

```

I prefer this setting when debugging APIs however the pretty print does not seem to work.  Is that a bug or a "feature"?

```json
{
"httpyac.responseViewContent": "exchange"
}

```

**Comment by AnWeber** [comment]
Rather something for which I have not yet found time and is also not quite simple. The reason is that for *.http files there is no [DocumentFormattingEditProvider](https://code.visualstudio.com/api/references/vscode-api#languages.registerDocumentFormattingEditProvider) registered. This would have to be able to handle different formatting specifications at the same time (JSON, Javascript, XML, ...). In principle, one could certainly copy that from [vetur](https://github.com/vuejs/vetur/). 
Currently I rely completely on the formatting used by vscode, only for JSON I try to partially achieve a better format using JSON.stringify.

**Reply by chrismalek** [reply]
Ok thanks for the reply. 

I work with mostly xml all day and noticed that XML is not formatted for `"httpyac.responseViewContent": "body"` but your response makes me understand why.

I know rest-client handles the pretty printing. I am guessing there are some implementation differences between how this library and that do the rendering so maybe a similar approach is not available.

To be honest, I don't know **anything** about the vscode extension development space and I am just a humble user of extensions provided by the community.

**Reply by AnWeber** [reply]
Thanks for the hint with rest-client. I found the [place that does the formatting](https://github.com/Huachao/vscode-restclient/blob/eead3c1ce5dec498ea96ac1fab8b0ac56bf83643/src/utils/responseFormatUtility.ts#L31). I create an issue for the function.

**Reply by AnWeber** [reply]
created issue #56

**Reply by chrismalek** [reply]
You rock!

**Reply by AnWeber** [reply]
But it will be the economy version, which only works for the output. It will not be able to format the self-written Http files. But I think that will help you in your case.

**Comment by AnWeber** [comment, marked as answer]
I have now implemented the automatic XML formatting with version 2.21.0. I use [xml-formatter](https://www.npmjs.com/package/xml-formatter) for formatting xml content.

**Comment by chrismalek** [comment]
This pretty printer was working on XML for a while. In the last few days it seems to have broken. I need to go through the latest release and figure out what changed.

**Reply by AnWeber** [reply]
I got reported  out of memory error with large files. I wanted to limit max pretty print size to 1Mb but just made 1Kb. Got an issue for that in httpyac repo and fixed it already. Will release it soon.

**Comment by chrismalek** [comment]
AnWeber..you rock...love it!   :-)

**Reply by AnWeber** [reply]
I just released version 4.3.0 which fixes this error.
